### ISSUE TITLE
Do not allocate RuntimeAssumptionTable on JITServer

### ISSUE THREAD

**Issue by mpirvu** [eclipse/openj9#8029]
JITServer should create no runtime assumptions, instead it should buffer them and send them to the client. In order to catch all mistakes, it's better to not even allocate a RuntimeAssumptionTable (generate a crash early).

**Comment by mpirvu** [comment]
Attn: @harryyu1994

**Comment by mpirvu** [comment]
For each assumption kind there is a hashtable of type `TR_RatHT`.
`env/RuntimeAssumptionTable.hpp` has 
```
class TR_RuntimeAssumptionTable
   {
   ...
   // My table of hash tables; init() allocate memory and populate it
   TR_RatHT _tables[LastAssumptionKind];
   }
```
In `rossa.cpp` in `onLoadInternal` we call
```
   if (!persistentMemory->getPersistentInfo()->getRuntimeAssumptionTable()->init())
      return -1;
```
and `init` is implemented in `runtime/ClassUloasAssumption.cpp` 
```
// Must call this during bootstrap on a single thread because it is not MT safe
 bool TR_RuntimeAssumptionTable::init()
    {
    ...
    for (int i=0; i < LastAssumptionKind; i++)
       {
       ...
       _tables[i]._htSpineArray = (OMR::RuntimeAssumption**)TR_PersistentMemory::jitPersistentAlloc(storageSize);
       _tables[i]._markedforDetachCount = (uint32_t*)TR_PersistentMemory::jitPersistentAlloc(sizeof(uint32_t)*_tables[i]._spineArraySize);
       if (!_tables[i]._htSpineArray || !_tables[i]._markedforDetachCount)
          return false;
       memset(_tables[i]._htSpineArray, 0, storageSize);
       memset(_tables[i]._markedforDetachCount, 0, sizeof(uint32_t)*_tables[i]._spineArraySize);
       }
...
```
We probably want to set NULL for those `_htSpineArray` instead of allocating memory.

**Comment by mpirvu** [comment]
Of course, we need to know that we are in JITServer mode

**Comment by mpirvu** [comment]
Fixed by #9189
